bftengine/src/preprocessor/PreProcessor.hpp

```diff
@@ -135,20 +135,14 @@ class PreProcessor {
 
   ~PreProcessor();
 
-  static void addNewPreProcessor(std::shared_ptr<MsgsCommunicator> &msgsCommunicator,
-                                 std::shared_ptr<IncomingMsgsStorage> &incomingMsgsStorage,
-                                 std::shared_ptr<MsgHandlersRegistrator> &msgHandlersRegistrator,
-                                 bftEngine::IRequestsHandler &requestsHandler,
-                                 InternalReplicaApi &myReplica,
-                                 concordUtil::Timers &timers,
-                                 std::shared_ptr<concord::performance::PerformanceManager> &pm);
-
-  static void setAggregator(std::shared_ptr<concordMetrics::Aggregator> aggregator);
-
   // For testing purposes
   ReqId getOngoingReqIdForClient(uint16_t clientId, uint16_t reqOffsetInBatch);
   RequestsBatchSharedPtr getOngoingBatchForClient(uint16_t clientId);
 
+  concordMetrics::Component &getMetricsComponent() { return metricsComponent_; }
+
+  concordUtil::RawMemoryPool &getMemoryPool() { return memoryPool_; }
+
```

Comment:
Where those two are used?

Reply:
`void ReplicaInternal::start() {
  if (ReplicaConfig::instance().getpreExecutionFeatureEnabled() && replica_->getAggregator() && preprocessor_) {
    preprocessor_->getMetricsComponent().SetAggregator(replica_->getAggregator());
    preprocessor_->getMemoryPool().setAggregator(replica_->getAggregator());
  }`

Reply:
It will be much better to expose PreProcessor function that sets an aggregator instead of exposing the metrics component and the pool.

Reply:
You are still exposing those functions.

Reply:
Done